code/game/objects/structures/benchpress.dm

```diff
@@ -0,0 +1,125 @@
+	icon = 'icons/obj/structures/benchpress.dmi'
+	icon_state = "benchpress_0"
+	base_icon_state = "benchpress"
+	density = FALSE
+	anchored = TRUE
+	pixel_x = -17
+	blocks_emissive = EMISSIVE_BLOCK_UNIQUE
+	var/datum/looping_sound/benchpress_creak/creak_loop
+	///amount of plates we are using, used to keep track for animations
+	var/plates = 0
+
+/obj/structure/benchpress/Initialize(mapload)
+	. = ..()
+	creak_loop = new
+	update_icon()
+
+/obj/structure/benchpress/update_icon_state()
+	. = ..()
+	icon_state = HAS_TRAIT(src, BENCH_BEING_USED) ? "[base_icon_state]_u" : "[base_icon_state]_[plates]"
+
+/obj/structure/benchpress/update_overlays()
+	. = ..()
+
+	if(HAS_TRAIT(src, BENCH_BEING_USED))
+		. += mutable_appearance(icon, "[base_icon_state]_[plates]_anim", plane = GAME_PLANE, layer = ABOVE_MOB_LAYER, alpha = src.alpha)
+
+/obj/structure/benchpress/attack_hand_alternate(mob/living/user)
+	. = ..()
+	if(.)
+		return
+	if(HAS_TRAIT(src, BENCH_BEING_USED))
+		return
+	if(plates)
+		var/oldplates = plates
+		plates = 0
+		update_icon()
+		flick("unswap_[oldplates]", src)
+		return
+	var/list/radial_options = list(
+		"1" = image(icon = 'icons/obj/structures/benchpress.dmi', icon_state = "benchpress_1"),
+		"2" = image(icon = 'icons/obj/structures/benchpress.dmi', icon_state = "benchpress_2"),
+		"3" = image(icon = 'icons/obj/structures/benchpress.dmi', icon_state = "benchpress_3"),
+		"4" = image(icon = 'icons/obj/structures/benchpress.dmi', icon_state = "benchpress_4"),
+		"5" = image(icon = 'icons/obj/structures/benchpress.dmi', icon_state = "benchpress_5"),
+	)
+	var/choice = show_radial_menu(user, src, radial_options, null, 64, null, TRUE, TRUE)
+	plates = text2num(choice)
+	update_icon()
+	flick("swap_[plates]", src)
+
+
+
+/obj/structure/benchpress/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+	if(HAS_TRAIT(src, BENCH_BEING_USED))
+		balloon_alert(user, "wait your turn!")
+		return
+	ADD_TRAIT(src, BENCH_BEING_USED, WEIGHTBENCH_TRAIT) // yea this is meh but IN_USE and interact code are a mess rn and too buggy so less sidestep it
+	update_icon()
+	user.setDir(SOUTH)
+	user.flags_atom |= DIRLOCK
+	ADD_TRAIT(user, TRAIT_IMMOBILE, WEIGHTBENCH_TRAIT)
+	user.forceMove(loc)
+	var/bragmessage = pick("pushing it to the limit","going into overdrive","burning with determination","rising up to the challenge", "getting strong now","getting ripped")
+	user.visible_message("<B>[user] is [bragmessage]!</B>")
+	addtimer(CALLBACK(src, PROC_REF(finish_press), user), 50)
+	creak_loop.start(src)
+
+///cleans up releases exerciser
+/obj/structure/benchpress/proc/finish_press(mob/user)
+	creak_loop.stop(src)
+	playsound(user, 'sound/machines/click.ogg', 60, TRUE)
+	REMOVE_TRAIT(src, BENCH_BEING_USED, WEIGHTBENCH_TRAIT)
+	user.flags_atom &= ~DIRLOCK
+	REMOVE_TRAIT(user, TRAIT_IMMOBILE, WEIGHTBENCH_TRAIT)
+	if(plates >= 5 && prob(10))
```

Comment:
number of plates making literally no difference except to break your arms is kinda funny.

Reply:
should be some progression and benefits. But that wasn't discussed thoroughly

Reply:
Shrug someone can add some later if they think of it